drivers/ethernet/eth_stm32_hal.c

```diff
@@ -38,6 +39,15 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include "eth.h"
 #include "eth_stm32_hal_priv.h"
 
+#if CONFIG_ETH_STM32_HAL_MAC3 || CONFIG_ETH_STM32_HAL_MAC4 || CONFIG_ETH_STM32_HAL_MAC5
+#define ETH_STM32_KCONFIG_MAC
+#warning CONFIG_ETH_STM32_HAL_MAC* is deprecated, use device tree "local-mac-address"
```

Comment:
This line is not needed with the choice, cmake will emit a warning if either of the 2 deprecated choices are selected